### PR TITLE
Added Samplingfeaturedataset endpoint. Added compatibility with pyodbc

### PR DESCRIPTION
**Pull request by Elijahwalkerwest**
This PR relates to #19.

### CODE REVIEW

**Comment by lsetiawan** [comment]
@Elijahwalkerwest There are many conflicts, please resolve.

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,7 @@
 .ipynb_checkpoints
 .DS_Store
 package.json
-node_modules
\ No newline at end of file
+node_modules
+development.py
```
Comment (lsetiawan) [inline .gitignore]: we might still be developing `development.py` so please remove from .gitignore

File: odm2rest/settings/base.py [hunk 2]
```diff
@@ -39,7 +39,6 @@
     'django.contrib.staticfiles',
     # External apps
     'rest_framework',
-    'rest_framework_swagger',
```
Comment (lsetiawan) [inline odm2rest/settings/base.py]: Thanks for catching this! Great.

File: api/urls.py [hunk 3]
```diff
@@ -54,6 +55,13 @@
         SamplingFeaturesViewSet.as_view(), name='samplingfeatures-detail'),
     url(r'^samplingfeatures/samplingFeatureType/(?P<samplingFeatureType>.+)/$',
         SamplingFeaturesViewSet.as_view(), name='samplingfeatures-detail'),
+    # Sampling Feature Datasets
+    url(r'^samplingfeaturedatasets/samplingFeatureID/(?P<samplingFeatureID>.+)/$',
```
Comment (lsetiawan) [inline api/urls.py]: These should be changed to query parameters now. Thanks.

**Comment by Elijahwalkerwest** [comment]
Did you want me to alter the function to accept query parameters? Or did you intend to do that for each? I pushed as such with the assumption you wanted to implement consistently across endpoints. 

Other than that, conflicts resolved, gitignore fixed.

**Comment by lsetiawan** [comment]
Go ahead and fix this one only within this PR.

**Comment by lsetiawan** [comment]
@Elijahwalkerwest For future PR, please create PR from your own forked repository rather than a branch in the Master repository to keep things cleaner. Thanks.

**Comment by Elijahwalkerwest** [comment]
Sorry about that, forgot to fork! Different work flow than I'm used to, will fix it for the next one. 

All should be good here, let me know if there are any problems.

File: api/views.py [hunk 4]
```diff
@@ -183,6 +186,37 @@ def get(self, request, samplingFeatureID=None,
...
+class SamplingFeaturesDataSetViewSet(APIView):
+
+    renderer_classes = (JSONRenderer, YAMLRenderer, CSVRenderer)
+
+    def get(self, request, samplingFeatureID=None,
+            samplingFeatureCode=None, samplingFeatureUUID=None,
+            dataSetType=None,):
+
+        get_kwargs = {
+            'samplingFeatureID': request.GET.get('samplingFeatureID'),
```
Comment (lsetiawan) [inline api/views.py]: please use `request.query_params.get('actionID')` instead.

Comment (Elijahwalkerwest) [reply api/views.py]: Done.

**Comment by lsetiawan** [comment]
Seems like you haven't update the `swagger.yaml` to show the `samplingfeaturedatasets`, is that correct?

**Comment by lsetiawan** [comment]
> Sorry about that, forgot to fork! Different work flow than I'm used to, will fix it for the next one.

No worries. Usually on a project, unless you are part of the core team, you won't be able to create new branches on the master repo, so one have to fork, make changes there and then create a PR. That's how you can contribute to any libraries out there.

**Comment by lsetiawan** [comment]
@Elijahwalkerwest Please address https://github.com/ODM2/ODM2RESTfulWebServices/pull/27#issuecomment-343597132 ASAP so I can merge this PR. Thanks.

**Comment by lsetiawan** [comment]
@Elijahwalkerwest What's the status of this PR? Have you addressed https://github.com/ODM2/ODM2RESTfulWebServices/pull/27#issuecomment-343597132? There are no new commits from you since https://github.com/ODM2/ODM2RESTfulWebServices/pull/27/commits/c2473f255422f6115e58eaa074d7fcc689e0281b.

**Comment by emiliom** [comment]
@Elijahwalkerwest and @lsetiawan, thanks for your work on this PR and the rest of the REST API.

@Elijahwalkerwest it'd be really helpful if you could move forward with this PR. @horsburgh and the rest of the ODM2 gang would like to have ODM2 REST API be in a reasonably useful form by the time of our workshop (starts Wednesday morning). Nice progress is being made, and this PR is an important part. Thanks!

**Comment by Elijahwalkerwest** [comment]
Sorry, haven't actually worked with Swagger before now. Will commit in a few minutes here, been making sure I have it right.

**Comment by emiliom** [comment]
> Sorry, haven't actually worked with Swagger before now. Will commit in a few minutes here, been making sure I have it right.

Great, thanks!

**Comment by Elijahwalkerwest** [comment]
Hey, added it to the swagger.yaml file, but I haven't worked with it before, and even though my swagger.json is good, I can't get it to show up in Swagger. Any ideas?

File: api/swagger/swagger.yaml [hunk 5]
```diff
@@ -161,6 +161,24 @@ paths:
...
           "200":
             description: 200 Response
+  /samplingfeaturedatasets/:
+    get:
+      tags:
+      - samplingfeaturedatasets
+      operationId: getSamplingFeatureDatasets
+      summary: Get ODM2 Datasets for given set of SamplingFeatures
+      parameters:
+        - name: samplingFeatureID
```
Comment (lsetiawan) [inline api/swagger/swagger.yaml]: each parameter needs an `in: query`

File: api/swagger/swagger.yaml [hunk 6]
```diff
@@ -161,6 +161,24 @@ paths:
         responses:
           "200":
             description: 200 Response
+  /samplingfeaturedatasets/:
+    get:
+      tags:
+      - samplingfeaturedatasets
```
Comment (lsetiawan) [inline api/swagger/swagger.yaml]: This tag `samplingfeaturedatasets` is not available, you have to specify this tag first up above.

**Comment by lsetiawan** [comment]
@Elijahwalkerwest Please look at instruction 4 & 5 in the [README](https://github.com/ODM2/ODM2RESTfulWebServices/blob/master/README.md) in the future when editing the `swagger.yml`. Thanks.

**Comment by lsetiawan** [comment]
> When I check the swagger.json file that swagger tells me it's reading, it isn't reflective of the code I have, as if none of the changes I am making are reflected.

It didn't work because you weren't meeting its schema. Don't worry about swagger.json. That file should be auto generated from the yaml file, so all you need is to edit the yaml, and making sure you are meeting the schema for swagger. Please read up on it here: https://swagger.io/specification/

**Comment by lsetiawan** [comment]
@Elijahwalkerwest Please take note of my changes on https://github.com/ODM2/ODM2RESTfulWebServices/pull/27/commits/6f0bc9ebae4e2e6affe2731f0161601be2e2948e. This now should meet the API specification, using query parameters and meeting the Swagger Specification. Thanks.

**Pull request merged**